parcels/grid.py

```diff
@@ -0,0 +1,142 @@
+from parcels.loggers import logger
+import numpy as np
+from ctypes import Structure, c_int, c_float, c_double, POINTER, cast, c_char_p, c_void_p, pointer
+from enum import IntEnum
+
+__all__ = ['GridCode', 'StructuredGrid', 'GridIndex', 'CGrid']
+
+
+class GridCode(IntEnum):
+    StructuredGrid = 0
+    SemiStructuredGrid = 1
+
+
+class CGrid(Structure):
+    _fields_ = [('name', c_char_p),
+                ('gtype', c_int),
+                ('grid', c_void_p)]
+
+
+class Grid(object):
+    """Grid class that defines a (spatial and temporal) grid on which Fields are defined
+
+    """
+
+    @property
+    def ctypes_struct(self):
+        self.cgrid = cast(pointer(self.child_ctypes_struct), c_void_p)
+        cstruct = CGrid(self.name, self.gtype, self.cgrid.value)
+        return cstruct
+
+
+class StructuredGrid(Grid):
+    """Structured Grid
+
+    :param name: Name of the field
+    :param lon: Longitude coordinates of the field
+    :param lat: Latitude coordinates of the field
+    :param depth: Depth coordinates of the field
+    :param time: Time coordinates of the field
+    """
+
+    def __init__(self, name, lon, lat, depth=None, time=None):
+        assert isinstance(lon, np.ndarray), 'lon is not a numpy array'
+        sh = lon.shape
+        assert(len(sh) == 1 or len(sh) == 2 and min(sh) == 2), 'lon is not a vector'
```

Comment:
Why do we need to assert whether `lon.shape` is a vector? Can't we do multi-dimensional arrays yet? Is that not the purpose of this PR?

Reply:
In this structured grid, all dimensions are vectors. Arrays will be defined for other types of grid.